packages/coding-style/src/Rector/ClassMethod/AnnotateThrowables.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodingStyle\Rector\ClassMethod;
+
+use PhpParser\Comment\Doc;
+use PhpParser\Node;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\Stmt\Throw_;
+use Rector\AttributeAwarePhpDoc\Ast\PhpDoc\AttributeAwareGenericTagValueNode;
+use Rector\AttributeAwarePhpDoc\Ast\PhpDoc\AttributeAwarePhpDocTagNode;
+use Rector\BetterPhpDocParser\PhpDocInfo\PhpDocInfo;
+use Rector\Core\Rector\AbstractRector;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+use Rector\Core\RectorDefinition\CodeSample;
+use Rector\Core\RectorDefinition\RectorDefinition;
+
+/**
+ * Adds "throws" DocBlock to methods.
+ */
+final class AnnotateThrowables extends AbstractRector
+{
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes() : array
+    {
+        return [Throw_::class];
+    }
+
+    /**
+     * From this method documentation is generated.
+     */
+    public function getDefinition() : RectorDefinition
+    {
+        return new RectorDefinition(
+            'Adds @throws DocBlock comments to methods that thrwo \Throwables.', [
+               new CodeSample(
+                    // code before
+                   '',
+                   // code after
+                   ''
+               ),
+           ]
+        );
+    }
+
+    /**
+     * @param Throw_ $node
+     *
+     * @return Node|null
+     */
+    public function refactor(Node $node) : ?Node
+    {
+        if ($this->isThrowableAnnotated($node)) {
+            return null;
+        }
+
+        $this->annotateMethod($node);
+
+        return $node;
+    }
+
+    /**
+     * @param Throw_ $node
+     *
+     * @return bool
+     */
+    private function isThrowableAnnotated(Throw_ $node) : bool
+    {
+        $method = $node->getAttribute(AttributeKey::METHOD_NODE);
+        $throwableParts = $node->expr->class->parts;
+        $docComment = $method->getDocComment();
+        $comments = $method->getComments();
+
+        if (null === $docComment && empty($comments)) {
+            return false;
+        }
+
+        return $this->isFQNAnnotated($docComment, $comments, $throwableParts);
+    }
+
+    /**
+     * @param Doc $docComment
+     * @param array<Doc> $comments
+     * @param array<string> $throwableParts
+     *
+     * @return bool
+     */
+    private function isFQNAnnotated(Doc $docComment, array $comments, array $throwableParts):bool
+    {
+        $FQN = $this->buildFQN($throwableParts);
+        $pattern = $this->buildThrowsDocComment($FQN);
+
+        if ($this->isDocCommentAlreadyPresent($docComment, $pattern)) {
+            return  true;
+        }
+
+        foreach ($comments as $comment) {
+            if ($this->isDocCommentAlreadyPresent($comment, $pattern)) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
You can use `PhpDocInfo` here too:

```php
/** @var PhpDocInfo $phpDocInfo */
        $phpDocInfo = $throw->getAttribute(AttributeKey::PHP_DOC_INFO);
        $throwTags = $phpDocInfo->getTagsByName('throws');

        /** @var ThrowsTagValueNode $throwTag */
        foreach ($throwTags as $throwTag) {
            // compare...
            $throwTag->type 
        }
        
```

Reply:
Ok, I'm going to work just on this to fix the tests.